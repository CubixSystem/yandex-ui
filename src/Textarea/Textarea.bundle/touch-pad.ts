import { compose, composeU, ExtractProps } from '@bem-react/core';

import { Textarea as TextareaTouchPad } from '../Textarea@touch-pad';
import { withDebounceInput } from '../../withDebounceInput';
// _hasClear
import { withHasClear } from '../_hasClear/Textarea_hasClear@touch-pad';
// _size
import { withSizeM } from '../_size/Textarea_size_m';
import { withSizeS } from '../_size/Textarea_size_s';
// _theme
import { withThemeNormal } from '../_theme/Textarea_theme_normal';
// _view
import { withViewDefault } from '../_view/Textarea_view_default';

export * from '../Textarea@touch-pad';

export const Textarea = compose(
    withDebounceInput,
    withHasClear,
    withViewDefault,
    withThemeNormal,
    composeU(withSizeM, withSizeS),
)(TextareaTouchPad);

export type ITextareaProps = ExtractProps<typeof Textarea>;
