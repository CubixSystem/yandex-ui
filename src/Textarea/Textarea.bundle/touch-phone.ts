import { compose, composeU, ExtractProps } from '@bem-react/core';

import { Textarea as TextareaTouchPhone } from '../Textarea@touch-phone';
import { withDebounceInput } from '../../withDebounceInput';
// _hasClear
import { withHasClear } from '../_hasClear/Textarea_hasClear@touch-phone';
// _size
import { withSizeM } from '../_size/Textarea_size_m';
import { withSizeS } from '../_size/Textarea_size_s';
// _theme
import { withThemeNormal } from '../_theme/Textarea_theme_normal';
// _view
import { withViewDefault } from '../_view/Textarea_view_default';

export * from '../Textarea@touch-phone';

export const Textarea = compose(
    withDebounceInput,
    withHasClear,
    withViewDefault,
    withThemeNormal,
    composeU(withSizeM, withSizeS),
)(TextareaTouchPhone);

export type ITextareaProps = ExtractProps<typeof Textarea>;
