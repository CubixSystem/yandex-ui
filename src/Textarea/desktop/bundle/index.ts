import { compose, composeU, ExtractProps } from '@bem-react/core';

import { Textarea as TextareaDesktop } from '../../Textarea@desktop';
import { withDebounceInput } from '../../../withDebounceInput';
// _hasClear
import { withHasClear } from '../../_hasClear/Textarea_hasClear@desktop';
// _size
import { withSizeM } from '../../_size/Textarea_size_m';
import { withSizeS } from '../../_size/Textarea_size_s';
// _theme
import { withThemeNormal } from '../../_theme/Textarea_theme_normal@desktop';
// _view
import { withViewDefault } from '../../_view/Textarea_view_default@desktop';
// _autoResize
import { withAutoResize } from '../../_autoResize/Textarea_autoResize';

export * from '../../Textarea@desktop';

export const Textarea = compose(
    withDebounceInput,
    withHasClear,
    withViewDefault,
    withThemeNormal,
    composeU(withSizeM, withSizeS),
)(TextareaDesktop);

export const TextareaWithAutoResize = withAutoResize(Textarea);

export type ITextareaProps = ExtractProps<typeof Textarea>;
