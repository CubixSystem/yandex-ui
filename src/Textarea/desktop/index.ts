export * from '../Textarea@desktop';
export * from '../_autoResize/Textarea_autoResize@desktop';
export * from '../_hasClear/Textarea_hasClear@desktop';
export * from '../_size/Textarea_size_m';
export * from '../_size/Textarea_size_s';
export * from '../_theme/Textarea_theme_normal@desktop';
export * from '../_view/Textarea_view_default@desktop';
