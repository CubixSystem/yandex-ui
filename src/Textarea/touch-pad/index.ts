export * from '../Textarea@touch-pad';
export * from '../_autoResize/Textarea_autoResize@touch-pad';
export * from '../_hasClear/Textarea_hasClear@touch-pad';
export * from '../_size/Textarea_size_m';
export * from '../_size/Textarea_size_s';
export * from '../_theme/Textarea_theme_normal';
export * from '../_view/Textarea_view_default';
