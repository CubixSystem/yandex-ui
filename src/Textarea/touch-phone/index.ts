export * from '../Textarea@touch-phone';
export * from '../_autoResize/Textarea_autoResize@touch-phone';
export * from '../_hasClear/Textarea_hasClear@touch-phone';
export * from '../_size/Textarea_size_m';
export * from '../_size/Textarea_size_s';
export * from '../_theme/Textarea_theme_normal';
export * from '../_view/Textarea_view_default';
