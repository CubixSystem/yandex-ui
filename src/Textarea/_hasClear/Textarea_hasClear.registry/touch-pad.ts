import { TextareaClear } from '../../Clear/Textarea-Clear';
import { textareaRegistry as textareaRegistryBase } from '../../Textarea.registry/touch-pad';

export const textareaRegistry = textareaRegistryBase.set('Clear', TextareaClear);
