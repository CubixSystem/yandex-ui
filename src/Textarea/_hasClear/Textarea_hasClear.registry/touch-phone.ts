import { TextareaClear } from '../../Clear/Textarea-Clear';
import { textareaRegistry as textareaRegistryBase } from '../../Textarea.registry/touch-phone';

export const textareaRegistry = textareaRegistryBase.set('Clear', TextareaClear);
