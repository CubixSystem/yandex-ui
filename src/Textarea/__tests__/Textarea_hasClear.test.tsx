import React, { createRef } from 'react';

import { createClientRender, fireEvent } from '../../internal/testing';
import { Textarea as TextareaDesktop, withHasClear as withHasClearDesktop } from '../desktop';
import { Textarea as TextareaTouchPad, withHasClear as withHasClearTouchPad } from '../touch-pad';
import { Textarea as TextareaTouchPhone, withHasClear as withHasClearTouchPhone } from '../touch-phone';

const platforms = [
    ['desktop', withHasClearDesktop(TextareaDesktop)],
    ['touch-pad', withHasClearTouchPad(TextareaTouchPad)],
    ['touch-phone', withHasClearTouchPhone(TextareaTouchPhone)],
] as const;

describe.each(platforms)('Textarea_hasClear@%s', (_platform, Textarea) => {
    const render = createClientRender();

    test('должен вернуть полный шаблон компонента (snapshot)', () => {
        const { container } = render(<Textarea hasClear />);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('должен вызывать событие onChange с пустым значением в event при клике в clear', () => {
        const onChange = jest.fn();
        const { container } = render(<Textarea hasClear onChange={onChange} />);

        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        fireEvent.click(container.querySelector('.Textarea-Clear')!);

        expect(onChange).toHaveBeenCalledTimes(1);
        expect(onChange.mock.calls[0][0].target.value).toBe('');
    });

    test('должен вызывать событие onClearClick при клике в clear', () => {
        const onClearClick = jest.fn();
        const { container } = render(<Textarea hasClear onClearClick={onClearClick} />);

        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        fireEvent.click(container.querySelector('.Textarea-Clear')!);

        expect(onClearClick).toHaveBeenCalledTimes(1);
    });

    test('должен проставлять фокус в textarea после клика в clear', () => {
        const controlRef = createRef<HTMLTextAreaElement>();
        const { container } = render(<Textarea hasClear controlRef={controlRef} />);

        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        fireEvent.click(container.querySelector('.Textarea-Clear')!);

        expect(controlRef.current).toEqual(document.activeElement);
    });
});
