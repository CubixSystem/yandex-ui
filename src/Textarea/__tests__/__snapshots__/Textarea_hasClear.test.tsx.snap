// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea_hasClear@desktop должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Textarea Textarea_hasClear"
>
  <span
    aria-hidden="true"
    class="Icon Icon_type_cross Textarea-Clear"
  />
  <span
    class="Textarea-Wrap"
  >
    <textarea
      aria-invalid="false"
      class="Textarea-Control"
    />
    <span
      class="Textarea-Box"
    />
  </span>
</span>
`;

exports[`Textarea_hasClear@touch-pad должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Textarea Textarea_hasClear"
>
  <span
    aria-hidden="true"
    class="Icon Icon_type_cross Textarea-Clear"
  />
  <span
    class="Textarea-Wrap"
  >
    <textarea
      aria-invalid="false"
      class="Textarea-Control"
    />
    <span
      class="Textarea-Box"
    />
  </span>
</span>
`;

exports[`Textarea_hasClear@touch-phone должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Textarea Textarea_hasClear"
>
  <span
    aria-hidden="true"
    class="Icon Icon_type_cross Textarea-Clear"
  />
  <span
    class="Textarea-Wrap"
  >
    <textarea
      aria-invalid="false"
      class="Textarea-Control"
    />
    <span
      class="Textarea-Box"
    />
  </span>
</span>
`;
